nixos/modules/services/networking/url-bot-rs.nix

```diff
@@ -0,0 +1,130 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.url-bot-rs;
+
+  format = pkgs.formats.toml {};
+  configFile = format.generate "url-bot-rs.toml" cfg.settings;
+in
+{
+  options.services.url-bot-rs = {
+    enable = mkEnableOption "Minimal IRC URL bot in Rust";
+
+    extraArgs = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = [ "--verbose" ];
+      description = ''
+        List of additional command line parameters for url-bot-rs.
+      '';
+    };
+
+    settings = mkOption {
+      type = format.type;
+      default = {};
+      description = ''
+        Configuration for url-bot-rs, see https://github.com/nuxeh/url-bot-rs#configuration-file-options
+        for supported values.
+      '';
+    };
+
+    nickname = mkOption {
+      type = types.str;
+      default = "url-bot-rs";
+      example = "urlify";
+      description = ''
+        Primary nickname of the bot on IRC.
+      '';
+    };
+
+    server = mkOption {
+      type = types.str;
+      example = "irc.example.com";
+      description = ''
+        Hostname of the IRC server to connect to.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 6667;
+      example = 6697;
+      description = ''
+        Port of the IRC server to connect to.
+      '';
+    };
+
+    ssl = mkOption {
+      type = types.bool;
+      default = cfg.port == 6697;
+      description = ''
+        Whether to enable SSL/TLS on the IRC connection.
+      '';
+    };
+
+    channels = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = ["#nixos"];
+      description = ''
+        List of channels the bot joins on connect.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.url-bot-rs.settings = {
+      connection = mkDefault {
```

Comment:
`connection = mapAttrs (name: mkDefault) {` to save some bytes?

Reply:
Yeah, bytes. It also looks alot cleaner. Thanks!